### PR TITLE
[WFLY-13807] EE: Add component class info to interface based proxy to preserve validation metadata

### PR DESCRIPTION
**Pull request by chengfang**
JIRA: https://issues.redhat.com/browse/WFLY-13807

This PR supercedes https://github.com/wildfly/wildfly/pull/13526, and fixed the 2 test failures in #13526

### CODE REVIEW

File: ee/src/main/java/org/jboss/as/ee/component/ViewService.java [hunk 1]
```diff
@@ -193,11 +195,24 @@ public Object invoke(InterceptorContext interceptorContext) throws Exception {
...
             return interceptor.processInvocation(interceptorContext);
         }
 
+        private Method matchMethod(Method invokedMethod) {
+            if (invokedMethod.getDeclaringClass().isInterface()) {
+                for (Method beanMethod : proxyFactory.getCachedMethods()) {
+                    if (MethodIdentifier.getIdentifierForMethod(beanMethod).equals(MethodIdentifier.getIdentifierForMethod(invokedMethod))) {
+                        return beanMethod;
+                    }
+                }
```
Comment (bstansberry) [inline ee/src/main/java/org/jboss/as/ee/component/ViewService.java:209]: This is on the request path and should be optimized to a single map lookup.

Comment (bstansberry) [reply ee/src/main/java/org/jboss/as/ee/component/ViewService.java:209]: I was referring to the entire for loop.

Comment (chengfang) [reply ee/src/main/java/org/jboss/as/ee/component/ViewService.java]: This path is only executed when `invokedMethod.getDeclaringClass().isInterface()` is true, i.e., when the bean is 2.x ejb, or the ejb bean class has no default constructor. So for the majority of ejb use case, this part is not touched. So I updated this part to populate a map on-demand to save the mapping.

Comment (bstansberry) [reply ee/src/main/java/org/jboss/as/ee/component/ViewService.java]: Nice; I like that approach.

**Review by bstansberry — APPROVED** [review]

**Pull request merged**